main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -344,6 +344,23 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
       }
     }
 
+    object Chars {
+      def Special: Rule1[String] = rule {
+        "\\\\" ~ push("\\") |
+          "\\'" ~ push("'") |
+          "\\\"" ~ push("\"") |
+          "\\n" ~ push("\n") |
+          "\\r" ~ push("\r") |
+          "\\t" ~ push("\t")
+      }
+
+      def Unicode: Rule1[String] = rule {
+        "\\u" ~ capture(4 times CharPredicate.HexDigit) ~> ((x: String) =>
+          // Convert the 4-digit string to a single character.
+          Integer.parseInt(x, 16).toChar.toString)
```

Comment:
Could this fail? If I type some illegal sequences of digits? I am afraid to get an exception during parsing.

Reply:
I don't think it could. If `Integer.parseInt` fails, then `x` must contain some non-hex character. But then `Charpredicate.HexDigit` would not match, so the parser wouldn't get that far.

Reply:
OK, and is any 4 digit hex a valid UTF8 char?

Reply:
Ah, I hadn't thought of that. I'll look into it.

Reply:
We could just do the conversion inside Weeder. So parse it as sequence of hex digits or similar.

Reply:
I'm thinking we might want to do all of the escape conversions inside the weeder. It would give use much nicer error messages.

Reply:
Also agreed!